pyproject.toml

```diff
@@ -39,106 +39,126 @@ server = [
 ]
 test = [
   "black==23.12.1",
+  "codespell==2.2.6",
   "flake8==7.0.0",
   "flake8-docstrings==1.7.0",
   "isort==5.13.2",
   "mypy==1.8.0",
+  "pre-commit==3.6.0",
+  "pre-commit-hooks==4.5.0",
   "pylint==3.0.3",
   "pytest==7.4.4",
   "pytest-aiohttp==1.0.5",
-  "pytest-cov==4.1.0",
+  "ruff==0.2.1",
+  "safety==3.0.1"
 ]
 
 [project.scripts]
 matter-server = "matter_server.server.__main__:main"
 
-[tool.codespell]
-ignore-words-list = "pase,"
+[tool.setuptools]
+platforms = ["any"]
+zip-safe  = false
+include-package-data = true
+
+[tool.setuptools.package-data]
+matter_server = ["py.typed"]
+
+[tool.setuptools.packages.find]
+include = ["matter_server*"]
 
 [tool.mypy]
-python_version = "3.10"
+platform = "linux"
+python_version = "3.11"
+
+# show error messages from unrelated files
+follow_imports = "normal"
+
+# suppress errors about unsatisfied imports
+ignore_missing_imports = true
+
+# be strict
 check_untyped_defs = true
-#disallow_any_generics = true
+disallow_any_generics = true
 disallow_incomplete_defs = true
+disallow_subclassing_any = true
 disallow_untyped_calls = true
+disallow_untyped_decorators = true
 disallow_untyped_defs = true
-mypy_path = "matter_server/"
 no_implicit_optional = true
-show_error_codes = true
+no_implicit_reexport = true
+strict_optional = true
 warn_incomplete_stub = true
+warn_no_return = true
 warn_redundant_casts = true
 warn_return_any = true
-warn_unreachable = true
 warn_unused_configs = true
 warn_unused_ignores = true
 
-[[tool.mypy.overrides]]
-ignore_missing_imports = true
-module = [
-  "aiorun",
-  "chip.*",
-  "coloredlogs",
+[tool.pylint.MASTER]
+extension-pkg-whitelist = [
+  "orjson",
+]
+ignore = [
+  "tests",
 ]
-
-[tool.pytest.ini_options]
-asyncio_mode = "auto"
-
-[tool.setuptools]
-platforms = ["any"]
-zip-safe  = false
-include-package-data = true
-
-[tool.setuptools.package-data]
-matter_server = ["py.typed"]
-
-[tool.setuptools.packages.find]
-include = ["matter_server*"]
-
 
 [tool.pylint.BASIC]
-class-const-naming-style = "any"
 good-names = [
-    "_",
-    "ev",
-    "ex",
-    "fp",
-    "i",
-    "id",
-    "j",
-    "k",
-    "Run",
-    "ip",
+  "_",
+  "ex",
```

Comment:
I don't think any of these are good names, besides `_` and possibly `T` as there some convention around those.